# Microsoft Studios Movie Project


# 1. Introduction

## 1.1 Overview

Films are a major part of popular culture and a huge source of entertainment for many people. The market is projected to reach USD 409.02 billion by 2026 in terms of revenue. Therefore, it is wise to invest in this market. I will use data from various websites that contain information about the film industry.

## 1.2 Objectives

1. Business Understanding<br>
2. Data Understanding <br>
3. Data Preparation
4. Data Analysis
5. Conclusion

# 2. Business Understanding

The tech giant Microsoft has decided to venture into creating original video content and wants to establish a movie studio. My goal is to use exploratory data analysis to produce insights for Microsoft as they enter the film industry. I will be looking for answers to the following questions:

1.What are the most profitable genres, franchises, and stars in each market and how do they differ across regions?

2.How do critical and audience ratings and reviews affect the box office performance of different films and how can they be used to predict demand and optimize marketing strategies?

3.What are the current and future challenges and opportunities in the film industry and how can Microsoft leverage its strengths and resources to create original video content that meets the need of the users?


# 3. Data Understanding

The datasets provided for this analysis were collected from different movie review aggregation sites and contain information on the various movie genres and their popularity among critics and viewers.  
The datasets include:
1. [IMDB](https://www.imdb.com/) 
2. [Box Office Mojo](https://www.boxofficemojo.com/) 
3. [Rotten Tomatoes](https://www.rottentomatoes.com/)


## Steps
1. Load the data with pandas and explore the dataframes.
2. Clean the data by dealing with:
    - missing values
    - duplicate rows
    - invalid data
    - outliers
3. Perform exploratory analysis in order to answer the business questions.
4. Conclusion.
5. Recommendations.

# 4. Loading Libraries and Datasets

In [3]:
# importing the packages I will be using for this project
import numpy as np
import sqlite3
import pandas as pd
import zipfile
import csv
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns

In [31]:
#loading the datasets
#first extract the im.db file and create a database connection
zf = zipfile.ZipFile('zippedData/im.db.zip')
zf.extract('im.db')
conn = sqlite3.connect('im.db')
#Loading other datasets
rt_reviews = pd.read_csv('zippedData/rt.reviews.tsv.gz',delimiter = "\t",encoding='latin-1')
rt_movies = pd.read_csv('zippedData/rt.movie_info.tsv.gz',delimiter = '\t')
bom_movies = pd.read_csv('zippedData/bom.movie_gross.csv.gz')
movie_budgets = pd.read_csv('zippedData/tn.movie_budgets.csv.gz')


# 3.2 Previewing the Datasets

#### a. Bom_movies

In [11]:
#previewing the top
bom_movies.head()

title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

  foreign_gross  year  
0     652000000  2010  
1     691300000  2010  
2     664300000  2010  
3     535700000  2010  
4     513900000  2010

In [13]:
#previewing the bottom 
bom_movies.tail()

title      studio  domestic_gross foreign_gross  \
3382                    The Quake       Magn.          6200.0           NaN   
3383  Edward II (2018 re-release)          FM          4800.0           NaN   
3384                     El Pacto        Sony          2500.0           NaN   
3385                     The Swan  Synergetic          2400.0           NaN   
3386            An Actor Prepares       Grav.          1700.0           NaN   

      year  
3382  2018  
3383  2018  
3384  2018  
3385  2018  
3386  2018

In [14]:
#determining the number of rows and columns
bom_movies.shape

(3387, 5)

In [15]:
#previewing bom_movies information
bom_movies.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3387 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3387 non-null   object 
 1   studio          3382 non-null   object 
 2   domestic_gross  3359 non-null   float64
 3   foreign_gross   2037 non-null   object 
 4   year            3387 non-null   int64  
dtypes: float64(1), int64(1), object(3)
memory usage: 132.4+ KB


In [16]:
#previewing the summary statistics of bom_movies
bom_movies.describe()

domestic_gross         year
count    3.359000e+03  3387.000000
mean     2.874585e+07  2013.958075
std      6.698250e+07     2.478141
min      1.000000e+02  2010.000000
25%      1.200000e+05  2012.000000
50%      1.400000e+06  2014.000000
75%      2.790000e+07  2016.000000
max      9.367000e+08  2018.000000

Observations:

1.The mean domestic gross is about 28.7 million dollars, with a large standard deviation of about 67 million dollars. The minimum domestic gross is 100 dollars and the maximum is about 936.7 million dollars.

2.The DataFrame has 3387 rows and 5 columns.

3.The info method shows that there are some missing values in the studio, domestic_gross, and foreign_gross columns

4.The foreign_gross column is of the object data type, which suggests that it may contain non-numeric values.

#### b. Rt_movies

In [17]:
#previewing the top of rt_movies
rt_movies.head()

id                                           synopsis rating  \
0   1  This gritty, fast-paced, and innovative police...      R   
1   3  New York City, not-too-distant-future: Eric Pa...      R   
2   5  Illeana Douglas delivers a superb performance ...      R   
3   6  Michael Douglas runs afoul of a treacherous su...      R   
4   7                                                NaN     NR   

                                 genre          director  \
0  Action and Adventure|Classics|Drama  William Friedkin   
1    Drama|Science Fiction and Fantasy  David Cronenberg   
2    Drama|Musical and Performing Arts    Allison Anders   
3           Drama|Mystery and Suspense    Barry Levinson   
4                        Drama|Romance    Rodney Bennett   

                            writer  theater_date      dvd_date currency  \
0                   Ernest Tidyman   Oct 9, 1971  Sep 25, 2001      NaN   
1     David Cronenberg|Don DeLillo  Aug 17, 2012   Jan 1, 2013        $   
2                   Allison Anders  Sep 13, 1996  Apr 18, 2000      NaN   
3  Paul Attanasio|Michael Crichton   Dec 9, 1994  Aug 27, 1997      NaN   
4                     Giles Cooper           NaN           NaN      NaN   

  box_office      runtime             studio  
0        NaN  104 minutes                NaN  
1    600,000  108 minutes  Entertainment One  
2        NaN  116 minutes                NaN  
3        NaN  128 minutes                NaN  
4        NaN  200 minutes                NaN

In [18]:
#previewing the bottom of rt_movies
rt_movies.tail()

id                                           synopsis rating  \
1555  1996  Forget terrorists or hijackers -- there's a ha...      R   
1556  1997  The popular Saturday Night Live sketch was exp...     PG   
1557  1998  Based on a novel by Richard Powell, when the l...      G   
1558  1999  The Sandlot is a coming-of-age story about a g...     PG   
1559  2000  Suspended from the force, Paris cop Hubert is ...      R   

                                                  genre            director  \
1555   Action and Adventure|Horror|Mystery and Suspense                 NaN   
1556                 Comedy|Science Fiction and Fantasy        Steve Barron   
1557  Classics|Comedy|Drama|Musical and Performing Arts      Gordon Douglas   
1558    Comedy|Drama|Kids and Family|Sports and Fitness  David Mickey Evans   
1559  Action and Adventure|Art House and Internation...                 NaN   

                                                writer  theater_date  \
1555                                               NaN  Aug 18, 2006   
1556  Terry Turner|Tom Davis|Dan Aykroyd|Bonnie Turner  Jul 23, 1993   
1557                                               NaN   Jan 1, 1962   
1558                  David Mickey Evans|Robert Gunter   Apr 1, 1993   
1559                                        Luc Besson  Sep 27, 2001   

          dvd_date currency  box_office      runtime             studio  
1555   Jan 2, 2007        $  33,886,034  106 minutes    New Line Cinema  
1556  Apr 17, 2001      NaN         NaN   88 minutes  Paramount Vantage  
1557  May 11, 2004      NaN         NaN  111 minutes                NaN  
1558  Jan 29, 2002      NaN         NaN  101 minutes                NaN  
1559  Feb 11, 2003      NaN         NaN   94 minutes  Columbia Pictures

In [20]:
#determining the number of rows and columns
rt_movies.shape

(1560, 12)

In [21]:
#previewing information on rt_movies
rt_movies.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1560 entries, 0 to 1559
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   id            1560 non-null   int64 
 1   synopsis      1498 non-null   object
 2   rating        1557 non-null   object
 3   genre         1552 non-null   object
 4   director      1361 non-null   object
 5   writer        1111 non-null   object
 6   theater_date  1201 non-null   object
 7   dvd_date      1201 non-null   object
 8   currency      340 non-null    object
 9   box_office    340 non-null    object
 10  runtime       1530 non-null   object
 11  studio        494 non-null    object
dtypes: int64(1), object(11)
memory usage: 146.4+ KB


Observations:

1. The currency, box_office, and studio columns have the most missing values, with only 340, 340, and 494 non-null entries respectively.
2. Values in genre column is separated by a vertical bar ( | ).
3. Runtime column is in string format I will drop the word "minutes" in the runtime column because it is in string format for easier analysis.

#### c. Rt_reviews

In [23]:
#previewing the top of rt_reviews
rt_reviews.head()

id                                             review rating   fresh  \
0   3  A distinctly gallows take on contemporary fina...    3/5   fresh   
1   3  It's an allegory in search of a meaning that n...    NaN  rotten   
2   3  ... life lived in a bubble in financial dealin...    NaN   fresh   
3   3  Continuing along a line introduced in last yea...    NaN   fresh   
4   3             ... a perverse twist on neorealism...     NaN   fresh   

           critic  top_critic         publisher               date  
0      PJ Nabarro           0   Patrick Nabarro  November 10, 2018  
1  Annalee Newitz           0           io9.com       May 23, 2018  
2    Sean Axmaker           0  Stream on Demand    January 4, 2018  
3   Daniel Kasman           0              MUBI  November 16, 2017  
4             NaN           0      Cinema Scope   October 12, 2017

In [24]:
#previewing the bottom of rt_reviews
rt_reviews.tail()

id                                             review rating   fresh  \
54427  2000  The real charm of this trifle is the deadpan c...    NaN   fresh   
54428  2000                                                NaN    1/5  rotten   
54429  2000                                                NaN    2/5  rotten   
54430  2000                                                NaN  2.5/5  rotten   
54431  2000                                                NaN    3/5   fresh   

                  critic  top_critic        publisher                date  
54427      Laura Sinagra           1    Village Voice  September 24, 2002  
54428  Michael Szymanski           0       Zap2it.com  September 21, 2005  
54429       Emanuel Levy           0  EmanuelLevy.Com       July 17, 2005  
54430   Christopher Null           0   Filmcritic.com   September 7, 2003  
54431    Nicolas Lacroix           0     Showbizz.net   November 12, 2002

In [25]:
#determining the number of rows and columns
rt_reviews.shape

(54432, 8)

In [26]:
#previewing information on rt_reviews
rt_reviews.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 54432 entries, 0 to 54431
Data columns (total 8 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   id          54432 non-null  int64 
 1   review      48869 non-null  object
 2   rating      40915 non-null  object
 3   fresh       54432 non-null  object
 4   critic      51710 non-null  object
 5   top_critic  54432 non-null  int64 
 6   publisher   54123 non-null  object
 7   date        54432 non-null  object
dtypes: int64(2), object(6)
memory usage: 3.3+ MB


In [28]:
#previewing the summary statistics
rt_reviews.describe()

id    top_critic
count  54432.000000  54432.000000
mean    1045.706882      0.240594
std      586.657046      0.427448
min        3.000000      0.000000
25%      542.000000      0.000000
50%     1083.000000      0.000000
75%     1541.000000      0.000000
max     2000.000000      1.000000

Observations:

1. The reviews seem to be sentences so they will not be of much use.
2. rt_movies and rt_reviews can be merged since they contain similar information.
3. The columns review,rating and critic appear have missing values  

#### d. movie_budgets

In [32]:
#previewing the top movie_budgets

movie_budgets.head()

id  release_date                                        movie  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

  production_budget domestic_gross worldwide_gross  
0      $425,000,000   $760,507,625  $2,776,345,279  
1      $410,600,000   $241,063,875  $1,045,663,875  
2      $350,000,000    $42,762,350    $149,762,350  
3      $330,600,000   $459,005,868  $1,403,013,963  
4      $317,000,000   $620,181,382  $1,316,721,747

In [33]:
#determining the number of rows and columns
movie_budgets.shape

(5782, 6)

In [34]:
#previewing information on movie_budgets
movie_budgets.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   id                 5782 non-null   int64 
 1   release_date       5782 non-null   object
 2   movie              5782 non-null   object
 3   production_budget  5782 non-null   object
 4   domestic_gross     5782 non-null   object
 5   worldwide_gross    5782 non-null   object
dtypes: int64(1), object(5)
memory usage: 271.2+ KB


Observations:
1. The data has 2 types : integer and object.
2. The columns includes id, rerlease_date, production_budget, domestic_gross and worldwide_gross

#### e. IMBD Movies

In [36]:
#viewing the list of tables in the imdb database
imdb_tables = pd.read_sql("""
SELECT name TableNames
FROM sqlite_master
WHERE type = 'table'
;
""",conn)
imdb_tables

TableNames
0   movie_basics
1      directors
2      known_for
3     movie_akas
4  movie_ratings
5        persons
6     principals
7        writers

In [50]:
#previewing the movie_basics table
imovie_basics = pd.read_sql_query('SELECT * FROM movie_basics', conn)
imovie_basics.head()

movie_id                    primary_title              original_title  \
0  tt0063540                        Sunghursh                   Sunghursh   
1  tt0066787  One Day Before the Rainy Season             Ashad Ka Ek Din   
2  tt0069049       The Other Side of the Wind  The Other Side of the Wind   
3  tt0069204                  Sabse Bada Sukh             Sabse Bada Sukh   
4  tt0100275         The Wandering Soap Opera       La Telenovela Errante   

   start_year  runtime_minutes                genres  
0        2013            175.0    Action,Crime,Drama  
1        2019            114.0       Biography,Drama  
2        2018            122.0                 Drama  
3        2018              NaN          Comedy,Drama  
4        2017             80.0  Comedy,Drama,Fantasy

In [39]:
#previewing the persons table
ipersons = pd.read_sql_query('SELECT * FROM persons', conn)
ipersons.head()

person_id       primary_name  birth_year  death_year  \
0  nm0061671  Mary Ellen Bauder         NaN         NaN   
1  nm0061865       Joseph Bauer         NaN         NaN   
2  nm0062070         Bruce Baum         NaN         NaN   
3  nm0062195       Axel Baumann         NaN         NaN   
4  nm0062798        Pete Baxter         NaN         NaN   

                                 primary_profession  
0         miscellaneous,production_manager,producer  
1        composer,music_department,sound_department  
2                        miscellaneous,actor,writer  
3  camera_department,cinematographer,art_department  
4  production_designer,art_department,set_decorator

In [48]:
#previewing the movie_ratings table
imovie_ratings = pd.read_sql_query('SELECT * FROM movie_ratings', conn)
imovie_ratings.head()

movie_id  averagerating  numvotes
0  tt10356526            8.3        31
1  tt10384606            8.9       559
2   tt1042974            6.4        20
3   tt1043726            4.2     50352
4   tt1060240            6.5        21

In [49]:
#previewing the directors table
idirectors = pd.read_sql_query('SELECT * FROM directors', conn)
idirectors.head()

movie_id  person_id
0  tt0285252  nm0899854
1  tt0462036  nm1940585
2  tt0835418  nm0151540
3  tt0835418  nm0151540
4  tt0878654  nm0089502

In [56]:
#I will join movie_ratings, directors and persons tables to movie_basics table since they all share movie_id and person_id.
imdb = pd.read_sql("""
SELECT *
FROM movie_basics b
JOIN movie_ratings r
USING (movie_id)
JOIN directors d
USING (movie_id)
JOIN persons p
USING (person_id)
;
""",conn)
imdb.head()

movie_id                    primary_title   original_title  start_year  \
0  tt0063540                        Sunghursh        Sunghursh        2013   
1  tt0063540                        Sunghursh        Sunghursh        2013   
2  tt0063540                        Sunghursh        Sunghursh        2013   
3  tt0063540                        Sunghursh        Sunghursh        2013   
4  tt0066787  One Day Before the Rainy Season  Ashad Ka Ek Din        2019   

   runtime_minutes              genres  averagerating  numvotes  person_id  \
0            175.0  Action,Crime,Drama            7.0        77  nm0712540   
1            175.0  Action,Crime,Drama            7.0        77  nm0712540   
2            175.0  Action,Crime,Drama            7.0        77  nm0712540   
3            175.0  Action,Crime,Drama            7.0        77  nm0712540   
4            114.0     Biography,Drama            7.2        43  nm0002411   

          primary_name  birth_year  death_year        primary_profession  
0  Harnam Singh Rawail      1921.0      2004.0  director,writer,producer  
1  Harnam Singh Rawail      1921.0      2004.0  director,writer,producer  
2  Harnam Singh Rawail      1921.0      2004.0  director,writer,producer  
3  Harnam Singh Rawail      1921.0      2004.0  director,writer,producer  
4            Mani Kaul      1944.0      2011.0     director,writer,actor

In [57]:
#closing the connection to the database
conn.close()

In [58]:
#previewing the top of the imdb dataframe
imdb.head()

movie_id                    primary_title   original_title  start_year  \
0  tt0063540                        Sunghursh        Sunghursh        2013   
1  tt0063540                        Sunghursh        Sunghursh        2013   
2  tt0063540                        Sunghursh        Sunghursh        2013   
3  tt0063540                        Sunghursh        Sunghursh        2013   
4  tt0066787  One Day Before the Rainy Season  Ashad Ka Ek Din        2019   

   runtime_minutes              genres  averagerating  numvotes  person_id  \
0            175.0  Action,Crime,Drama            7.0        77  nm0712540   
1            175.0  Action,Crime,Drama            7.0        77  nm0712540   
2            175.0  Action,Crime,Drama            7.0        77  nm0712540   
3            175.0  Action,Crime,Drama            7.0        77  nm0712540   
4            114.0     Biography,Drama            7.2        43  nm0002411   

          primary_name  birth_year  death_year        primary_profession  
0  Harnam Singh Rawail      1921.0      2004.0  director,writer,producer  
1  Harnam Singh Rawail      1921.0      2004.0  director,writer,producer  
2  Harnam Singh Rawail      1921.0      2004.0  director,writer,producer  
3  Harnam Singh Rawail      1921.0      2004.0  director,writer,producer  
4            Mani Kaul      1944.0      2011.0     director,writer,actor

In [59]:
#previewing the bottom of the imdb dataframe
imdb.tail()

movie_id               primary_title              original_title  \
181382  tt9914642                   Albatross                   Albatross   
181383  tt9914642                   Albatross                   Albatross   
181384  tt9914942  La vida sense la Sara Amat  La vida sense la Sara Amat   
181385  tt9914942  La vida sense la Sara Amat  La vida sense la Sara Amat   
181386  tt9916160                  Drømmeland                  Drømmeland   

        start_year  runtime_minutes       genres  averagerating  numvotes  \
181382        2017              NaN  Documentary            8.5         8   
181383        2017              NaN  Documentary            8.5         8   
181384        2019              NaN         None            6.6         5   
181385        2019              NaN         None            6.6         5   
181386        2019             72.0  Documentary            6.5        11   

        person_id        primary_name  birth_year  death_year  \
181382  nm5300859        Chris Jordan         NaN         NaN   
181383  nm5300859        Chris Jordan         NaN         NaN   
181384  nm1716653           Laura Jou         NaN         NaN   
181385  nm1716653           Laura Jou         NaN         NaN   
181386  nm5684093  Joost van der Wiel         NaN         NaN   

                     primary_profession  
181382           director,writer,editor  
181383           director,writer,editor  
181384   miscellaneous,actress,director  
181385   miscellaneous,actress,director  
181386  director,cinematographer,writer

In [60]:
#determining the number of rows and columns
imdb.shape

(181387, 13)

In [61]:
#checking the information
imdb.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 181387 entries, 0 to 181386
Data columns (total 13 columns):
 #   Column              Non-Null Count   Dtype  
---  ------              --------------   -----  
 0   movie_id            181387 non-null  object 
 1   primary_title       181387 non-null  object 
 2   original_title      181387 non-null  object 
 3   start_year          181387 non-null  int64  
 4   runtime_minutes     163584 non-null  float64
 5   genres              180047 non-null  object 
 6   averagerating       181387 non-null  float64
 7   numvotes            181387 non-null  int64  
 8   person_id           181387 non-null  object 
 9   primary_name        181387 non-null  object 
 10  birth_year          54805 non-null   float64
 11  death_year          1342 non-null    float64
 12  primary_profession  181262 non-null  object 
dtypes: float64(4), int64(2), object(7)
memory usage: 18.0+ MB


In [62]:
#previewing the summary statistics of imdb 
imdb.describe()

start_year  runtime_minutes  averagerating      numvotes  \
count  181387.000000    163584.000000  181387.000000  1.813870e+05   
mean     2014.309802        97.789484       6.217683  4.955524e+03   
std         2.536111       194.434689       1.388026  3.760931e+04   
min      2010.000000         3.000000       1.000000  5.000000e+00   
25%      2012.000000        84.000000       5.400000  1.900000e+01   
50%      2014.000000        94.000000       6.300000  6.600000e+01   
75%      2016.000000       107.000000       7.200000  3.110000e+02   
max      2019.000000     51420.000000      10.000000  1.841066e+06   

         birth_year   death_year  
count  54805.000000  1342.000000  
mean    1969.097856  2014.908346  
std       12.499740     4.866581  
min     1870.000000  1944.000000  
25%     1962.000000  2014.000000  
50%     1971.000000  2016.000000  
75%     1978.000000  2018.000000  
max     2004.000000  2019.000000

Observations:
1. The genres are separated by commas.
2. There are columns with missing data: runtime_minutes, genres, birth_year, death_year and primary_profession.
3. The longest movie is 51,420 minutes and the lowest 3 minutes long and the standard deviation is at 194.4 from runtime_minutes.
4. The highest number of votes cast was 1,841,066 and the lowest 5 only.

# 5. Data Cleaning

This section prepares the data for exploratory analysis. It handles the problems with the datasets that have been observed while loading and exploring them. It will cover:
   - Duplicated rows
   - Missing values
   - Unwanted observations
   - Structural issues
   - Outliers
   - Invalid data
   - Merges
   
